internal/delivery/http/v1/auth.go

```diff
@@ -66,13 +61,7 @@ func (h *Handler) userLogin(ctx *gin.Context) {
 	result, err := h.services.Users.Login(ctx.Request.Context(), &input)
 
 	if err != nil {
-		if err == service.ErrInvalidCredentials {
-			utils.ErrorResponse(ctx, http.StatusBadRequest, err)
-			return
-		}
-
-		utils.ErrorResponse(ctx, http.StatusInternalServerError, err)
-		return
+		h.handleServiceError(ctx, err)
 	}
 
```

Comment:
This method returns service.PostUserLoginOutput on success. Please update swagger. Also it makes sense using h.parseRequestBody() helper here too

Reply:
Addressed in the latest commit.